pkg/configmapandsecret/secretfactory.go

```diff
@@ -61,15 +65,20 @@ func (f *SecretFactory) MakeSecret(args *types.SecretArgs) (*corev1.Secret, erro
 	var err error
 	s := f.makeFreshSecret(args)
 
-	pairs, err := f.keyValuesFromEnvFileCommand(args.EnvCommand)
+	timeout := defaultCommandTimeout
```

Comment:
fwiw, its a little less code to slap a timeout field into the SecretFactory struct, and initialize it in the "constructor".  The override the field here if need be, and then not have to pass it to the various factory functions.  up to you.

Reply:
Well, single `SecretFactory`'s `MakeSecret()` can be called multiple times for different `SecretArgs` which may have different timeouts, so using `SecretFactory` state to avoid passing an extra arg to other methods doesn't seem to be correct approach. It'll work (except for possible race conditions factory funcs are invoked from different goroutines), but still I doubt this field belongs to the factory.